build.gradle

```diff
@@ -21,137 +21,185 @@ plugins {
+                "org.slf4j:log4j-over-slf4j",
+                "org.slf4j:slf4j-api",
+                "net.logstash.logback:logstash-logback-encoder",
+        ],
+        javax                 : [
+                "org.hibernate.javax.persistence:hibernate-jpa-2.1-api",
+                "javax.transaction:jta",
+                "javax.persistence:javax.persistence-api",
+                "javax.xml.bind:jaxb-api",
+        ],
+        commons               : [
+                "commons-codec:commons-codec",
+                "commons-collections:commons-collections",
+                "commons-io:commons-io",
+                "commons-lang:commons-lang",
+        ],
+        guice                 : [
+                "com.google.inject.extensions:guice-assistedinject",
+                "com.google.inject.extensions:guice-multibindings",
+                "com.google.inject.extensions:guice-servlet",
+                "com.google.inject.extensions:guice-throwingproviders",
+                "com.google.inject.extensions:guice-persist",
+                "com.google.inject:guice",
+                "aopalliance:aopalliance",
+                "javax.inject:javax.inject",
+        ],
+        liquibase             : "org.liquibase:liquibase-core",
+        liquibase_slf4j       : "com.mattbertolini:liquibase-slf4j",
+        oauth                 : [
+                "net.oauth.core:oauth",
+                "net.oauth.core:oauth-provider",
+        ],
+        collections           : "com.google.guava:guava:25.1-jre",
+        hibernate_validator_ap: "org.hibernate.validator:hibernate-validator-annotation-processor",
+        checkstyle            : [
+                "com.puppycrawl.tools:checkstyle",
+                "com.github.sevntu-checkstyle:sevntu-checks"
+        ],
+        gettext               : "com.googlecode.gettext-commons:gettext-commons",
+        javax_servlet         : "javax.servlet:servlet-api",
+        javax_validation      : "javax.validation:validation-api",
+        jmock                 : [
+                "org.jmock:jmock",
+                "org.jmock:jmock-junit4",
+        ],
+        validator             : [
+                "org.hibernate.validator:hibernate-validator",
+                "org.hibernate.validator:hibernate-validator-annotation-processor",
+        ],
+        keycloak              : [
+                "org.keycloak:keycloak-servlet-filter-adapter",
+                "org.keycloak:keycloak-adapter-core",
+                "org.keycloak:keycloak-common",
+                "org.keycloak:keycloak-core",
+                "org.keycloak:keycloak-adapter-spi",
+                "org.keycloak:keycloak-servlet-adapter-spi",
+                "org.apache.httpcomponents:httpcore",
+                "org.bouncycastle:bcpkix-jdk15on",
+                "org.bouncycastle:bcprov-jdk15on",
+                "org.keycloak:keycloak-jaxrs-oauth-client",
+                "org.keycloak:keycloak-authz-client",
```

Comment:
Please remove all but necessary. I suspect only keycloak-servlet-filter-adapter is necessary. I suggest starting with the obviously transitive deps and then removing and seeing if you can still build, until you reach a minimum set.

Reply:
Done.